entrypoint.sh

```diff
@@ -40,11 +41,34 @@ case "$connection_format" in
 esac
 
 # run copa to patch image
-if copa patch -i "$image" -r ./data/"$report" -t "$patched_tag" $connection --timeout $timeout $output;
+if [ "$max_retries" -eq 0 ]
 then
-    patched_image="$image_no_tag:$patched_tag"
-    echo "patched-image=$patched_image" >> "$GITHUB_OUTPUT"
+    if copa patch -i "$image" -r ./data/"$report" -t "$patched_tag" "$connection" --timeout "$timeout" "$output"
+    then
+        patched_image="$image_no_tag:$patched_tag"
+        echo "patched-image=$patched_image" >> "$GITHUB_OUTPUT"
+    else
+        echo "Error patching image $image with copa"
+        exit 1
+    fi
 else
-    echo "Error patching image $image with copa"
-    exit 1
-fi
+    retries=0
+    while [ "$retries" -lt "$max_retries" ]
+    do
+        if copa patch -i "$image" -r "./data/$report" -t "$patched_tag" "$connection" --timeout "$timeout" "$output"
+        then
+            patched_image="$image_no_tag:$patched_tag"
```

Comment:
do you think we could put this in a function since its repeated code?

Reply:
@ashnamehrotra Good catch :+1:. Yeah, it is always better to use functions in this case :smile:.